mlrun/runtimes/pod.py

```diff
@@ -338,6 +345,15 @@ def _verify_and_set_requests(
         patch: bool = False,
     ):
         resources = verify_requests(resources_field_name, mem=mem, cpu=cpu)
+        for pattern in mlrun.utils.regex.pipeline_param:
+            if re.match(pattern, str(cpu)) or re.match(pattern, str(mem)):
+                self.__fields_pending_rollback.update(
+                    {
+                        resources_field_name: copy.deepcopy(
+                            getattr(self, resources_field_name)
+                        ),
+                    }
+                )
```

Comment:
id move to `self._add_field_to_pending_discard(resources_field_name, getattr(self, resources_field_name))` 
that way you can
1. have a bit more concise function
2. if fields_pending_rollback structure is changed, no need to add in many places (assuming more function will reuse that logic

wdyt?

Reply:
Good thinking. I made this refactor now.